### PR TITLE
feat(frontend): support primary key of tables

### PR DESCRIPTION
**Pull request by wzzzzd**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Only insert `_row_id` column to the table if no primary key is provided. and it is inserted as the last column (for easier debugging). 
- Add optional `row_id_index` to protobuf message `StreamSourceInfo` and `TableSourceInfo`.
- Remove all assertions  and hard-code that `row_id_index == 0`.

NOTE: Currently we cannot handle duplicate values in pk columns. For example,
```sql
create table t (a int primary key);
insert into t values (1), (1);
```
This will cause a 'double insert' panic.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/4256
close https://github.com/singularity-data/risingwave/issues/4726

### CODE REVIEW

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM. 🚀🚀🚀 Would you please add some e2e tests for table with pks?

**Review by BugenZhao — APPROVED** [review]
LGTM. @st1page @xiangjinwu May you please also take a look? 🥰

File: proto/catalog.proto [hunk 1]
```diff
@@ -7,18 +7,24 @@ import "plan_common.proto";
...
-  int32 row_id_index = 4;
+  ColumnIndex row_id_index = 4;
   repeated plan_common.ColumnCatalog columns = 5;
   repeated int32 pk_column_ids = 6;
 }
 
 message TableSourceInfo {
-  repeated plan_common.ColumnCatalog columns = 1;
-  map<string, string> properties = 2;
+  ColumnIndex row_id_index = 1;
```
Comment (st1page) [inline proto/catalog.proto:24]: maybe this field should be in the `SourceExecutor` node? (we can do it later.) c.c. @BugenZhao

**Review by st1page — APPROVED** [review]
LGTM

File: proto/catalog.proto (hunk 1 above)
Comment (BugenZhao) [reply proto/catalog.proto:24]: Yes. Currently we assemble a `ProstSource` and then convert it to `SourceCatalog` to construct the `StreamSource` node, which is weird. We can refactor it later. 🥰

**Review by xiangjinwu — APPROVED** [review]
LGTM. Is it possible to run slt e2e tests?

**Pull request merged**